### ISSUE TITLE
earthaccess should start using CMR metadata to obtain S3 credentials

### ISSUE THREAD

**Issue by betolink** [nsidc/earthaccess#280]
`earthaccess` uses a hard-coded list of [DAAC](https://github.com/nsidc/earthaccess/blob/main/earthaccess/daac.py) endpoints to get temporary S3 credentials, however more and more datasets are under different endpoints and/or a different category( e.g. as services), Having `earthaccess` using the `DirectDistributionInformation/S3CredentialsAPIEndpoint` from the UMM metadata record would be a more robust solution. This could bring a time penalty so caching would also be nice to have.

**Comment by jhkennedy** [comment]
@scottstanie thanks! Yes, ASF has multiple endpoints now and the best place to determine which endpoint to use is the CMR metadata.

@betolink do you tink we could eliminate the hard coding entirely?

**Comment by scottstanie** [comment]
Thank you!
Also I realized right after I posted that, I can also use the `search_datasets` results to get credentials without picking a random product:
```python
In [43]: rs = earthaccess.search_datasets(daac='ASF', keyword="OPERA_L2_CSLC-S1_V1")
Datasets found: 1

In [44]: rs[0].s3_bucket()
Out[44]:
{'Region': 'us-west-2',
 'S3BucketAndObjectPrefixNames': ['asf-cumulus-prod-opera-products/OPERA_L2_CSLC-S1/',
  'asf-cumulus-prod-opera-browse/OPERA_L2_CSLC-S1/'],
 'S3CredentialsAPIEndpoint': 'https://cumulus.asf.alaska.edu/s3credentials',
 'S3CredentialsAPIDocumentationURL': 'https://cumulus.asf.alaska.edu/s3credentialsREADME'}
```

but I noticed that I have to do
```python
auth = earthaccess.login()
auth.get_s3_credentials(endpoint=...)
```
since the top level `earthaccess.get_s3_credentials` doesn't have the `endpoint` argument. Still seems easy enough for me though 👍

**Comment by betolink** [comment]
I think this was resolved in `v0.8.2`

I tested this and it worked as expected without the need to specify the endpoint (it will use the metadata from CMR):

```python
import earthaccess
import xarray as xr

earthaccess.login()

results = earthaccess.search_data(granule_name='OPERA_L2_CSLC-S1_T012-024520-IW1_20231007T124805Z_20231009T193746Z_S1A_VV_v1.0', concept_id="C2777443834-ASF")
file = earthaccess.open(results)[0]

ds = xr.open_dataset(file, group="data")
ds
```

if we want to download the file it would be the same:

```python
import earthaccess

earthaccess.login()

results = earthaccess.search_data(granule_name='OPERA_L2_CSLC-S1_T012-024520-IW1_20231007T124805Z_20231009T193746Z_S1A_VV_v1.0', concept_id="C2777443834-ASF")
files = earthaccess.download(results, "opera/")

```

**Comment by asteiker** [comment]
Resolved by #296
